lib/emerald/parser.rb

```diff
@@ -34,8 +36,13 @@ def parse_whitespace(source)
       [nil, rest_of_source]
     end
 
-    def parse_atom(source)
-      pattern = /\A[a-z\-]+/
+    def parse_atom(source, non_whitespace_char)
+      case non_whitespace_char
+      when /\A[a-zA-Z\-]+/
+        pattern = /\A[a-zA-Z\-]+/
+      when /\d+/
+        pattern = /\d+/
+      end
       atom_value = pattern.match(source).to_s
```

Comment:
I think it's a bit muddled because numbers and atoms are parsed by the same method. How about a `parse_atom` and `parse_number`? We already know if it's an atom or number in the function above so it knows what one to pick.

The latter could return an `Emerald::Number`